### PR TITLE
Debug Filter Feature, Add Edit Error Message

### PR DESCRIPTION
**Pull request by jowhee3011**
Modify filter command and parser
- Fix test cases

Add edit error message
- Fix test case

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 1]
```diff
@@ -33,17 +40,17 @@ public class FilterCommand extends Command {
...
+    private final Email email;
+    private final Phone phone;
+    private final Address address;
 
     /**
      * Creates a FilterCommand to filter persons by the specified {@code name}, {@code role}, {@code email},
      * {@code phone} and {@code address}.
      */
-    public FilterCommand(String name, String role, String email, String phone, String address) {
+    public FilterCommand(Name name, Optional<Role> role, Email email, Phone phone, Address address) {
```
Comment (Droas590) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:53]: Nice very robust

**Review by Droas590 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/FilterCommand.java [hunk 2]
```diff
@@ -102,11 +108,23 @@ public boolean equals(Object other) {
...
+
+        boolean equalEmail = (email == null && otherFilterCommand.email == null)
+                || email != null && email.equals(otherFilterCommand.email);
+
+        boolean equalPhone = (phone == null && otherFilterCommand.phone == null)
+                || phone != null && phone.equals(otherFilterCommand.phone);
+
+        boolean equalAddress = (address == null && otherFilterCommand.address == null)
+                || address != null && address.equals(otherFilterCommand.address);
+
```
Comment (SherniceSng) [inline src/main/java/seedu/address/logic/commands/FilterCommand.java:126]: Very thorough equals validation!

**Review by SherniceSng — COMMENTED** [review]
LGTM.

**Pull request merged**